src/clustering/administration/servers/auto_reconnect.cc

```diff
@@ -84,10 +84,25 @@ void auto_reconnector_t::try_reconnect(const server_id_t &server,
 
     wait_any_t interruptor(&reconnected, &give_up_timer, keepalive.get_drain_signal());
 
+    cond_t join_failed;
     exponential_backoff_t backoff(50, 15 * 1000);
     try {
-        while (!interruptor.is_pulsed()) {
-            connectivity_cluster_run->join(last_known_address, join_delay_secs);
+        while (!interruptor.is_pulsed() && !join_failed.is_pulsed()) {
+            coro_t::spawn_now_dangerously([&]() {
```

Comment:
This lambda function has to capture a copy of `keepalive` to make sure that the auto_reconnector_t is not destructed while the coroutine exists:
```
coro_t::spawn_sometime([keepalive, &]() {
```

I think with that keepalive copy, it can be a normal `spawn_sometime`, and doesn't need to be a `spawn_now_dangerously` anymore since the `connectivity_cluster_run` is guaranteed to remain valid for as long as the `auto_reconnector_t` instance is there.